From dff3e51e37d6bf2863afddaca63e2c3eb1e809af Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Tue, 15 Oct 2019 18:20:34 +0200
Subject: [PATCH 0054/1239] arm64: mvebu: armada-8040-mcbin: enable networking
 in device tree

Enabled interfaces:
10G SFI CP0, 10G SFI CP1, 2.5G SGMII CP1, 1G SGMII CP1;
1G SGMII CP1 is connected to CP0 mdio;
10G SFI CP0 and 10G SFI CP1 are connected to CP0 xmdio.

Change-Id: I9189babd873ee798729061ee7efe165fa5bbe696
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
---
 arch/arm/dts/armada-8040-mcbin.dts | 44 ++++++++++++++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/arch/arm/dts/armada-8040-mcbin.dts b/arch/arm/dts/armada-8040-mcbin.dts
index 23df369747..caf6544def 100644
--- a/arch/arm/dts/armada-8040-mcbin.dts
+++ b/arch/arm/dts/armada-8040-mcbin.dts
@@ -153,6 +153,7 @@
 };
 
 &cp0_mdio {
+	status = "okay";
 	ge_phy: ethernet-phy@0 {
 		reg = <0>;
 	};
@@ -188,6 +189,31 @@
 	};
 };
 
+&cp0_xmdio {
+	status = "okay";
+	sfi_phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+
+	sfi_phy8: ethernet-phy@8 {
+		reg = <8>;
+	};
+};
+
+&cp0_ethernet {
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+/* 10G SFI/copper */
+&cp0_eth0 {
+	status = "okay";
+	phy-mode = "sfi";
+	phy = <&sfi_phy0>;
+	phy-reset-gpios = <&cp1_gpio0 31 GPIO_ACTIVE_HIGH>;
+	marvell,sfp-tx-disable-gpio = <&cp1_gpio0 29 GPIO_ACTIVE_HIGH>;
+};
+
 &cp1_sata0 {
 	status = "okay";
 };
@@ -205,12 +231,30 @@
 	status = "okay";
 };
 
+/* 10G SFI/copper */
+&cp1_eth0 {
+	status = "okay";
+	phy-mode = "sfi";
+	phy = <&sfi_phy8>;
+	phy-reset-gpios = <&cp1_gpio0 9 GPIO_ACTIVE_HIGH>;
+	marvell,sfp-tx-disable-gpio = <&cp1_gpio0 10 GPIO_ACTIVE_HIGH>;
+};
+
+/* 1G SGMII */
 &cp1_eth1 {
 	status = "okay";
 	phy = <&ge_phy>;
 	phy-mode = "sgmii";
 };
 
+/* 2.5G SGMII */
+&cp1_eth2 {
+	status = "okay";
+	phy-mode = "sgmii";
+	phy-speed = <2500>;
+	marvell,sfp-tx-disable-gpio = <&cp1_gpio0 24 GPIO_ACTIVE_HIGH>;
+};
+
 &cp1_pinctl {
 	/*
 	 * MPP Bus:
-- 
2.29.0

